scripts/west_commands/run_common.py

```diff
@@ -142,90 +142,97 @@ def do_run_common(command, user_args, user_runner_args):
     logger.setLevel(LOG_LEVEL)
     logger.addHandler(WestLogHandler())
 
-    # If the user passed -- to force the parent argument parser to stop
-    # parsing, it will show up here, and needs to be filtered out.
-    runner_args = [arg for arg in user_runner_args if arg != '--']
+    # Load runners.yaml files.
+    runners_yaml = runners_yaml_path(cache)
 
-    # Arguments are provided in this order to allow the specific to
-    # override the general:
-    #
-    # - common runners.yaml arguments
-    # - runner-specific runners.yaml arguments
-    # - command line arguments
-    final_argv = (runner_config['args']['common'] +
-                  runner_config['args'][runner_name] +
-                  runner_args)
-
-    # 'user_args' contains parsed arguments which are:
-    #
-    # 1. provided on the command line, and
-    # 2. handled by add_parser_common(), and
-    # 3. *not* runner-specific
-    #
-    # 'final_argv' contains unparsed arguments from either:
-    #
-    # 1. runners.yaml, or
-    # 2. the command line
-    #
-    # We next have to:
-    #
-    # - parse 'final_argv' now that we have all the command line
-    #   arguments
-    # - create a RunnerConfig using 'user_args' and the result
-    #   of parsing 'final_argv'
-    parser = argparse.ArgumentParser(prog=runner_name)
-    add_parser_common(command, parser=parser)
-    runner_cls.add_parser(parser)
-    args, unknown = parser.parse_known_args(args=final_argv)
-    if unknown:
-        log.die(f'runner {runner_name} received unknown arguments: {unknown}')
-
-    # Override args with any user_args. The latter must take
-    # precedence, or e.g. --hex-file on the command line would be
-    # ignored in favor of a board.cmake setting.
-    for a, v in vars(user_args).items():
-        if v is not None:
-            setattr(args, a, v)
-
-    # Create the RunnerConfig from the values assigned to common
-    # arguments. This is a hacky way to go about this; probably
-    # ZephyrBinaryRunner should define what it needs to make this
-    # happen by itself. That would be a larger refactoring of the
-    # runners package than there's time for right now, though.
-    #
-    # Use that RunnerConfig to create the ZephyrBinaryRunner instance
-    # and call its run().
-    runner = runner_cls.create(runner_cfg_from_args(args, build_dir), args)
-    try:
-        runner.run(command_name)
-    except ValueError as ve:
-        log.err(str(ve), fatal=True)
-        dump_traceback()
-        raise CommandError(1)
-    except MissingProgram as e:
-        log.die('required program', e.filename,
-                'not found; install it or add its location to PATH')
-    except RuntimeError as re:
-        if not user_args.verbose:
-            log.die(re)
-        else:
-            log.err('verbose mode enabled, dumping stack:', fatal=True)
-            raise
+    for current_yaml in runners_yaml:
+        # A CMake list may contain multiple `;;` which can result in an empty
```

Comment:
This is being called from dump_context() too, but the filtering is not being done there. I think the filtering should be done in runners_yaml_path() instead to make sure it always happens.

```
def runners_yaml_path(cache):
    ret = cache.get_list('ZEPHYR_RUNNERS_YAML')

    if ret is None:
        board = cache.get('CACHED_BOARD')
        log.die(f'either a pristine build is needed, or board {board} '
                "doesn't support west flash/debug "
                '(no ZEPHYR_RUNNERS_YAML in CMake cache)')

    # A CMake list may contain multiple `;;` which can result in an empty
    # entry in the result. Thus, simply ignore empty elements using filter().
    return [yml for yml in ret if yml]
```

This is also switching to use get_list(). Without that, this didn't work for me using a vanilla zephyr build, where the result is a string.

Reply:
Will move.

Reply:
Just making sure you noticed it's not just a move; there is another semantic change:

> This is also switching to use get_list(). Without that, this didn't work for me using a vanilla zephyr build, where the result is a string.